DQMOffline/Trigger/plugins/TagAndProbeBtagTriggerMonitor.cc

```diff
@@ -0,0 +1,187 @@
+  discr_offline_btag_jet2_ = ibooker.book1D("discr_offline_btag_jet2","discr_offline_btag_jet2",btagnbins, btagbins);
+  
+  // Initialize the GenericTriggerEventFlag
+  if ( genTriggerEventFlag_ && genTriggerEventFlag_->on() ) genTriggerEventFlag_->initRun( iRun, iSetup );
+}
+
+void TagAndProbeBtagTriggerMonitor::analyze(edm::Event const& iEvent, edm::EventSetup const& iSetup)
+{
+
+//    bool accept = false;
+   bool match1 = false;
+   bool match2 = false;
+   
+   edm::Handle<reco::JetTagCollection> offlineJetTagPFHandler;
+   iEvent.getByToken(offlineBtagToken_, offlineJetTagPFHandler);
+      
+   if( ! offlineJetTagPFHandler.isValid()) return;
+   
+   // applying selection for event; tag & probe -> selection  for all events
+   if (genTriggerEventFlag_->on() && ! genTriggerEventFlag_->accept( iEvent, iSetup) )
+   {
+      
+      reco::JetTagCollection jettags = *(offlineJetTagPFHandler.product());
+      if ( jettags.size() > 1 )
+      {
+         const reco::Jet * jet1 = jettags.key(0).get();
+         const reco::Jet * jet2 = jettags.key(1).get();
+         TLorentzVector p4_jet1;
+         p4_jet1.SetPtEtaPhiM(jet1->pt(),jet1->eta(),jet1->phi(),0);
+         TLorentzVector p4_jet2;
+         p4_jet2.SetPtEtaPhiM(jet2->pt(),jet2->eta(),jet2->phi(),0);
+         float btag1 = jettags.value(0);
+         float btag2 = jettags.value(1);
+            
+         if ( jet1->pt() > jetPtmin_ && jet2->pt() > jetPtmin_ && fabs(jet1->eta()) < jetEtamax_ && fabs(jet2->eta()) < jetEtamax_ )
+         {
+            if ( btag1 > tagBtagmin_ && btag2 > probeBtagmin_ )
+            {
+               pt_jet1_  -> Fill(jet1->pt());
+               pt_jet2_  -> Fill(jet2->pt());
+               eta_jet1_ -> Fill(jet1->eta());
+               eta_jet2_ -> Fill(jet2->eta());
+               phi_jet1_ -> Fill(jet1->phi());
+               phi_jet2_ -> Fill(jet2->phi());
+               eta_phi_jet1_ -> Fill(jet1->eta(),jet1->phi());
+               eta_phi_jet2_ -> Fill(jet2->eta(),jet2->phi());
+               if ( btag1 < 0 ) btag1 = -0.5;
+               if ( btag2 < 0 ) btag2 = -0.5;
+               discr_offline_btag_jet1_ -> Fill(btag1);
+               discr_offline_btag_jet2_ -> Fill(btag2);
+               
+// trigger objects matching  
+               std::vector<trigger::TriggerObject> onlinebtags;           
+               edm::Handle<trigger::TriggerEvent> triggerEventHandler;
+               iEvent.getByToken(triggerSummaryToken_, triggerEventHandler);
+               const unsigned int filterIndex(triggerEventHandler->filterIndex(edm::InputTag(triggerobjbtag_,"",processname_)));
+               if ( filterIndex < triggerEventHandler->sizeFilters() )
+               {
+                  const trigger::Keys& keys(triggerEventHandler->filterKeys(filterIndex));
+                  const trigger::TriggerObjectCollection & triggerObjects = triggerEventHandler->getObjects();
+                  for ( auto & key : keys )
```

Comment:
please add onlinebtags.reserve(onlinebtags.size()+keys.size())
[maybe the syntax is different for finding the length of keys]